### PR TITLE
[SYCL][L0] Fix pinned memory allocation test.

### PR DESCRIPTION
**Pull request by rbegam**
Pinned memory allocation is not currently supported by L0.

Signed-off-by: rbegam <rehana.begam@intel.com>

### CODE REVIEW

File: SYCL/Basic/alloc_pinned_host_memory.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// REQUIRES: level_zero || cuda
+// REQUIRES: cuda
```
Comment (vladimirlaz) [inline SYCL/Basic/alloc_pinned_host_memory.cpp]: Does the test failing on Level_Zero satbily? If it is I would prefer to mark test as XFAIL: level_zero

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

**Comment by smaslov-intel** [comment]
@rbegam : why does it fail? I known we don't truly support pinned memory, but my expectation is that it is just a **nop** for Level-Zero for now. So why this is failing?

**Comment by rbegam** [comment]
@smaslov-intel, this test checks SYCL_PI_TRACE for zeMemAllocHost() (https://github.com/intel/llvm-test-suite/pull/215/files#diff-fb8c60f542fd5b66745807155566683a3d084c409b5375d523268bb46f06d222L43) while buffer create. This is failing for the discrete gpu as we don't call zeMemAllocHost() anymore. For integrated it's passing.

**Comment by smaslov-intel** [comment]
> @smaslov-intel, this test checks SYCL_PI_TRACE for zeMemAllocHost() (https://github.com/intel/llvm-test-suite/pull/215/files#diff-fb8c60f542fd5b66745807155566683a3d084c409b5375d523268bb46f06d222L43) while buffer create. This is failing for the discrete gpu as we don't call zeMemAllocHost() anymore. For integrated it's passing.

Ah, OK. Then I suggest that we remove the check for `zeMemAllocHost` (since we know now it is unrelated to pinning) and make this test to check that `{ext::oneapi::property::buffer::use_pinned_host_memory()}` is accepted.

**Comment by rbegam** [comment]
@smaslov-intel cuda has that call checked. let's separate the tests for level_zero (remove zeMemAllocHost) and cuda (as is).

File: SYCL/Basic/alloc_pinned_host_memory.cpp (hunk 1 above)
Comment (rbegam) [reply SYCL/Basic/alloc_pinned_host_memory.cpp]: modified the CHECK

**Comment by smaslov-intel** [comment]
> @smaslov-intel cuda has that call checked. let's separate the tests for level_zero (remove zeMemAllocHost) and cuda (as is).

But there is no `zeMemAllocHost` in CUDA. This is Level-Zero interface

**Review by smaslov-intel — APPROVED** [review]

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**